react/features/toolbox/components/web/Toolbox.js

```diff
@@ -941,13 +1100,88 @@ class Toolbox extends Component<Props> {
         const toolbarAccLabel = 'toolbar.accessibilityLabel.moreActionsMenu';
+        const buttonsLeft = [];
+        const buttonsRight = [];
+
+        const maxNumberOfButtonsPerGroup = Math.floor(
+            (
+                this.state.windowWidth
+                    - 168 // the width of the central group by design
+                    - 48 // the minimum space between the button groups
+            )
+            / 56 // the width + padding of a button
+            / 2 // devide by the number of groups(left and right group)
+        );
+
+        console.log(maxNumberOfButtonsPerGroup);
+
+        if (this._shouldShowButton('desktop')
+                && this._isDesktopSharingButtonVisible()) {
+            buttonsLeft.push('desktop');
+        }
+        if (this._shouldShowButton('raisehand')) {
+            buttonsLeft.push('raisehand');
+        }
+        if (this._shouldShowButton('chat')) {
+            buttonsLeft.push('chat');
+        }
+        if (this._shouldShowButton('closedcaptions')) {
+            buttonsLeft.push('closedcaptions');
+        }
+        if (overflowHasItems) {
+            buttonsRight.push('overflowmenu');
+        }
+        if (this._shouldShowButton('info')) {
+            buttonsRight.push('info');
+        }
+        if (this._shouldShowButton('invite') && !_hideInviteButton) {
+            buttonsRight.push('invite');
+        }
+        if (this._shouldShowButton('tileview')) {
+            buttonsRight.push('tileview');
+        }
+        if (this._shouldShowButton('localrecording')) {
+            buttonsRight.push('localrecording');
+        }
+
+        const movedButtons = [];
+
+        if (buttonsLeft.length > maxNumberOfButtonsPerGroup) {
+            movedButtons.push(...buttonsLeft.splice(
+                maxNumberOfButtonsPerGroup,
+                buttonsLeft.length - maxNumberOfButtonsPerGroup));
+            if (buttonsRight.indexOf('overflowmenu') === -1) {
+                buttonsRight.unshift('overflowmenu');
+            }
+        }
+
+        if (buttonsRight.length > maxNumberOfButtonsPerGroup) {
+            if (buttonsRight.indexOf('overflowmenu') === -1) {
+                buttonsRight.unshift('overflowmenu');
+            }
+
+            let numberOfButtons = maxNumberOfButtonsPerGroup;
+
+            // make sure the more button will be displayed when we move buttons.
+            if (numberOfButtons === 0) {
+                numberOfButtons++;
+            }
+
+            movedButtons.push(...buttonsRight.splice(
+                numberOfButtons,
+                buttonsRight.length - numberOfButtons));
+
+        }
+
+        overflowMenuContent.splice(
+            1, 0, ...this._renderMovedButtons(movedButtons));
 
         return (
             <div className = 'toolbox-content'>
                 <div className = 'button-group-left'>
-                    { this._shouldShowButton('desktop')
+                    { buttonsLeft.indexOf('desktop') !== -1
```

Comment:
Array.prototype.includes I think reads much better than indexOf.

Reply:
It does! But wasn't indexOf faster? I was thinking this render could be called many times... WDYT?

If you are not worried about the performance I'll change it!

Reply:
jsperf shows about 500 more operations a second over includes; fairly negligible and depends on chrome version. If you want to get into performance you should look at changing the buttons to a set.